jwst/lib/set_telescope_pointing.py

```diff
@@ -148,6 +144,77 @@ def update_wcs(model, default_pa_v3=0., siaf_path=None, **kwargs):
         Keyword arguments used by matrix calculation routines.
     """
 
+    # If the type of exposure is not FGS, then attempt to get pointing
+    # from telemetry.
+    if model.meta.exposure.type.lower().startswith('fgs'):
+        update_wcs_from_header(
+            model, default_pa_v3=default_pa_v3, siaf_path=siaf_path, **kwargs
+        )
+    else:
+        update_wcs_from_telem(
+            model, default_pa_v3=default_pa_v3, siaf_path=siaf_path, **kwargs
+        )
+
+
+def update_wcs_from_header(model, default_pa_v3=0., siaf_path=None, **kwargs):
+    """ Update WCS pointing from header information
+
+    For FGS-like observations, nearly all information is already populated
+    except for the CD matrix. This simply calculates that matrix.
```

Comment:
Could we include a more specific comment here - to facilitate future readings. These are FGS GUIDER observations.
We populate only the FITS WCS keywords.